tests/public_api/values.cpp

```diff
@@ -594,3 +594,23 @@ TEST_CASE("destructuring of Vallist") {
         CHECK(nil2 == minilua::Nil());
     }
 }
+
+TEST_CASE("next(table [, index]") {
+    SECTION("empty table") {
+        const minilua::Table value;
+        CHECK(value.next(minilua::Nil()) == minilua::Vallist());
+    }
+    SECTION("filled table") {
+        const minilua::Table value{{"key1", 22}, {1, "Hallo "}, {2, "Welt!"}, {100, 42}};
+        SECTION("access last element of table") {
+            CHECK(value.next(minilua::Value("key1")) == minilua::Vallist());
```

Comment:
You shouldn't rely on the order of the map being stable. You should use something like this

```cpp
*(--value.end())
```

to get the last entry.

Reply:
This doesn't work because the table-struct directly doesn't support the itterator-functions.

Reply:
We should only test behavior we actually want. In this case we want to ensure that `next` for the last element returns `Nil` (or an empty `Vallist`). We should add a method to table to get the last element.

Since adding proper `begin` and `end` without exposing how we implement `Table` is not easily possible in C++ we would need to implement an iterator our selves. Which we probably should do anyway to make it possible to iterate over the map in C++. But that task is too big for this PR.

I would just leave a TODO in the tests for the moment and I will open an issue to implement an iterator for `Table`.

Reply:
#63

Reply:
As it turns out maps in C++ only implement a forward iterator and not a bidirectional iterator. That means you can't decrement them.

The only other way I see of testing this is to call `next` on a key and then again on the result until you get `Nil`. If you don't get nil after 4 calls the test fails.